code/game/objects/structures/statues.dm

```diff
@@ -7,25 +7,81 @@
 
 /obj/structure/statue/Destroy()
 	density = 0
 	..()
 
-/obj/structure/statue/attackby(obj/item/weapon/W, mob/user)
-	hardness -= W.force/100
-	user << "You hit the [name] with your [W.name]!"
-	CheckHardness()
+/obj/structure/statue/attackby(obj/item/weapon/W, mob/living/user as mob)
+	add_fingerprint(user)
+	user.changeNext_move(CLICK_CD_MELEE)
+	if(istype(W, /obj/item/weapon/wrench))
+		if(anchored)
+			playsound(src.loc, 'sound/items/Ratchet.ogg', 100, 1)
+			user.visible_message("<span class='notice'>[user] is loosening the [name]'s bolts...</span>", \
+								 "<span class='notice'>You are loosening the [name]'s bolts...</span>")
+			if(do_after(user,40))
+				if(!src) return
+				if(!anchored) return
+				user.visible_message("<span class='notice'>[user] loosened the [name]'s bolts!</span>", \
+									 "<span class='notice'>You loosened the [name]'s bolts!</span>")
+				anchored = 0
+		else if(!anchored)
+			if (!istype(src.loc, /turf/simulated/floor))
+				user.visible_message("<span class='danger'>A floor must be present to secure the [name]!</span>")
+				return
+			playsound(src.loc, 'sound/items/Ratchet.ogg', 100, 1)
+			user.visible_message("<span class='notice'>[user] is securing the [name]'s bolts...</span>", \
+								 "<span class='notice'>You are securing the [name]'s bolts...</span>")
+			if(do_after(user, 40))
+				if(!src) return
+				if(anchored) return
+				user.visible_message("<span class='notice'>[user] has secured the [name]'s bolts!</span>", \
+									 "<span class='notice'>You have secured the [name]'s bolts!</span>")
+				anchored = 1
+
+	else if(istype(W, /obj/item/weapon/pickaxe/plasmacutter))
+		user.visible_message("<span class='notice'>[user] is slicing apart the [name]...</span>", \
+							 "<span class='notice'>You are slicing apart the [name]...</span>")
+		if(do_after(user,30))
+			if(!src.loc) return
+			user.visible_message("<span class='notice'>[user] slices apart the [name]!</span>", \
+								 "<span class='notice'>You slice apart the [name]!</span>")
+			Dismantle(1)
+
+	else if(istype(W, /obj/item/weapon/pickaxe/drill/diamonddrill))
+		user.visible_message("<span class='notice'>[user] begins to drill apart the [name]!</span>", \
+							 "<span class='notice'>You begin to drill apart the [name]!</span>")
+		if(do_after(user,5))
+			if(!src.loc) return
+			user.visible_message("<span class='notice'>[user] destroys the [name]!</span>", \
+								 "<span class='notice'>You destroy the [name]!</span>")
+			qdel(src)
+
+	else if(istype(W, /obj/item/weapon/weldingtool))
+		if(!anchored)
+			user.visible_message("<span class='notice'>[user] is slicing apart the [name]...</span>", \
+								 "<span class='notice'>You are slicing apart the [name]...</span>")
+			if(do_after(user, 40))
+				if(!src.loc) return
+				user.visible_message("<span class='notice'>[user] slices apart the [name]!</span>", \
+									 "<span class='notice'>You slice apart the [name]!</span>")
+				Dismantle(1)
+
+	else
```

Comment:
Can you remove that 'else' and add 'return' at the end of each action. It would make attacking an anchored statue with a welding tool possible.